### DISCUSSION TITLE
【任务】2023年3-4月社区激励任务｜50,000 $PEOPLE

### DISCUSSION THREAD

**Discussion by AmazingAng** [AmazingAng/WTF-Solidity#435] [任务 | Bounty]
WTF Academy 社区激励任务，包括但不限于审核，改善WTF极简教程，也包括日常的PR/issue。这部分奖励优先分配给月度社区激励任务，剩余分给当月贡献者。

## 任务 1: 校对WTF Solidity极简教程31-51讲的翻译 （领取完毕）

内容：WTF Solidity教程是用中文写的，为了让国际开发者可以使用，我们现在需要将教程翻译成英文。根据[Solidity 官方文档](https://docs.soliditylang.org/en/v0.8.17/)的英语名词，校对[WTF Solidity](https://github.com/AmazingAng/WTF-Solidity) 极简教程31-51讲的 chatgpt [翻译](https://github.com/AmazingAng/WTF-Solidity/blob/main/Languages/en/README.md)，包括教程文档，图片，代码注释。

要求：

- 学习过 WTF Solidity 教程
- 参与过 WTF Academy 社区之前的翻译任务 或 能证明自己的英语能力。
- 阅读过英文 Solidity 官方文档。

赏金：15,000 $PEOPLE 代币，根据工作量分配，每人最多同时认领 5 讲。
31-35: [Rulesbreaker](https://github.com/Rulesbreaker)  3571 $people 0x7fe42Fbb75320a109cdc9ffd32ea801fA570d626
36-40: [yeecai](https://github.com/yeecai) 3571 $people 0xb900d23ec39e668272BeA2D7122a004Bf1974d8A
41-46: [0x7869616F](https://github.com/0x7869616F)  0x1660E55837b8bb4932f29D316F75CdE80Cfa2DE8 3571 $people
46-51: [Azleal](https://github.com/Azleal) 4285 $people 0x8e779db3B2D3a7b3D9030e665BfBB27F703565E0

## 任务 2: 审核 WTF Ethers 教程的v6版本

内容：最近 Ethers.js 大版本更新，v5 -> v6，变化挺大，几乎所有之前的脚本都要改。目前文档还不完善，迁移成本较高。
WTF Ethers 教程已经更新到 v6 版本，但需要审核代码，教程正文和配图（1-24讲）。

要求：

- 学习过 WTF Solidity 和 WTF Ethers 教程
- 使用过 ethers.js v6
- 阅读过 ethers.js 还不完整的文档

赏金：10,000 $PEOPLE 代币，根据工作量分配，每人最多同时认领 5 讲。

1-5, 17-19: [Azleal](https://github.com/Azleal) 3333 $people 0x8e779db3B2D3a7b3D9030e665BfBB27F703565E0
6-10: [buttonwild](https://github.com/buttonwild) 2083 $people 0x430de06fA4B93ac1993Eafd4BF96b6A1e8F76a32
11-13: [cheng521521](https://github.com/cheng521521) 1250 $people DreamMoon.eth
14-16: [skyMetaverse](https://github.com/skyMetaverse) 1250 $people syskeys.eth
20-24: [hattiepwvi](https://github.com/hattiepwvi) 2083 $people 0xE968B97044b832d9577C79F10555396581F2E074

## 黑客松: 开源轻量solidity网页编译器

目标是完成一个基于react的轻便型solidity网页编译器，用它可以扩展出各种不同的文档插件和教程使用。这个项目会参与3月 ETH Global 举办的 Scaling Ethereum 黑客松，然后用于 wtf.academy 教育平台。

要求:

- 熟悉 solidity，ethersjs
- 熟悉 react 组件开发
- 熟悉工具库开发

仓库: 

https://github.com/WTFAcademy/solive

## 认领方式
在这个帖子下面回复就能认领，认领人需要填写过[WTF社区成员申请表格](https://docs.google.com/forms/d/e/1FAIpQLSe4KGT8Sh6sJ7hedQRuIYirOoZK_85miz3dw7vA1-YjodgJ-A/viewform?usp=sf_link)。

有用的资料：
1. Markdown书写规则（**必看**）：https://www.markdownguide.org/cheat-sheet/
4. 审核并添加图片后的例子：https://github.com/AmazingAng/WTFSolidity/tree/main/02_ValueTypes
5. 审核第一期：https://github.com/AmazingAng/WTFSolidity/discussions/18
6. https://github.com/AmazingAng/WTFSolidity/discussions/132
7. 8月社区激励任务：https://github.com/AmazingAng/WTF-Solidity/discussions/174

奖励已发放，发的映射到 BNB chain 的 [$PEOPLE](https://bscscan.com/token/0x2c44b726adf1963ca47af88b284c06f30380fc78)，可以在pancakeswap或币安swap。
TxHash: https://bscscan.com/tx/0x0863f611aa45a0661803935beabf9219618655b4f727a79e46e1dc01eb114c66
![截屏2023-06-14 上午11 38 48](https://github.com/AmazingAng/WTF-Solidity/assets/14728591/5ff24146-9b18-401c-b557-a36b5472eb17)

**Comment by Rulesbreaker** [comment]
申请领取31-35

**Reply by AmazingAng** [reply]
好的

**Comment by beyond009** [comment]
报名黑客松 熟悉solidity etherjs react

**Reply by AmazingAng** [reply]
好的

**Comment by chaoticpunk** [comment]


**Reply by AmazingAng** [reply]
收到

**Comment by jessecaii** [comment]
申请领取36-40

**Reply by AmazingAng** [reply]
收到

**Reply by jessecaii** [reply]
已完成，请问代币什么时候发放

**Reply by AmazingAng** [reply]
机翻之后需要人工审核一下名词，我留言到pr上了。
修改完成后这几天就发放。

**Comment by KieSun** [comment]
报名黑客松。

了解 solidity，熟悉 etherjs react 及工具库的开发。

**Comment by Azleal** [comment]
申请领取46-50

**Reply by AmazingAng** [reply]
好的

**Reply by Azleal** [reply]
@AmazingAng 想确认下任务描述中说的中文翻译的英文原文是什么? 这里的一讲好像不一定能在solidity英文文档中能找到。比如这里46讲代理合约的原始英文在哪里? 
还是说对照languages下的英文版本的来校对中文版本?

**Reply by Azleal** [reply]
> > @AmazingAng 想确认下任务描述中说的中文翻译的英文原文是什么? 这里的一讲好像不一定能在solidity英文文档中能找到。比如这里46讲代理合约的原始英文在哪里?
> > 还是说对照languages下的英文版本的来校对中文版本?
> 
> Proxy

感谢回复，我第一次参与这个任务有很多不明白的地方。
我还是对你说的不太理解。solidity文档中没有单独的一节讲的是proxy吧？我找到的和proxy最有关的是[这个](https://docs.soliditylang.org/en/v0.8.17/security-considerations.html?#authorized-proxies)
如果方便麻烦告知下你上面说的Proxy的具体链接，感谢~

**Reply by AmazingAng** [reply]
> @AmazingAng 想确认下任务描述中说的中文翻译的英文原文是什么? 这里的一讲好像不一定能在solidity英文文档中能找到。比如这里46讲代理合约的原始英文在哪里? 
> 还是说对照languages下的英文版本的来校对中文版本?

编辑了任务描述，让它更容易理解。WTF Solidity用中文编写，现在需要翻译成英文。

目前已经用chatgpt翻译成英文，需要校对。solidity文档里没有proxy也没关系，只要保证名词一致就行。

**Reply by Azleal** [reply]
> > @AmazingAng 想确认下任务描述中说的中文翻译的英文原文是什么? 这里的一讲好像不一定能在solidity英文文档中能找到。比如这里46讲代理合约的原始英文在哪里?
> > 还是说对照languages下的英文版本的来校对中文版本?
> 
> 编辑了任务描述，让它更容易理解。WTF Solidity用中文编写，现在需要翻译成英文。
> 
> 目前已经用chatgpt翻译成英文，需要校对。solidity文档里没有proxy也没关系，只要保证名词一致就行。

明白了👍🏻

**Comment by chiguayeshao** [comment]
认领黑客松，熟悉React，了解Solidity与Ethers.js

**Comment by Azleal** [comment]
申请领取51

**Reply by AmazingAng** [reply]
收到！

**Comment by cradle0fFilth** [comment]
"目标是完成一个基于react的轻便型solility网页编译器"这里写错了应该是solidity

**Comment by buttonwild** [comment]
认领任务2  6-10讲

**Reply by AmazingAng** [reply]
收到

**Comment by Azleal** [comment]
认领任务2  1-5讲

**Reply by AmazingAng** [reply]
收到

**Reply by Azleal** [reply]
已完成

**Comment by skyMetaverse** [comment]
认领任务2   14-16讲

**Reply by AmazingAng** [reply]
收到

**Comment by cheng521521** [comment]
认领任务2 11-13

**Reply by AmazingAng** [reply]
收到

**Comment by hattiepwvi** [comment]
申请认领20-24讲

**Reply by AmazingAng** [reply]
收到

**Comment by Azleal** [comment]
申请认领任务2 17-19讲

**Comment by tjgljs** [comment]
报名黑客松。
精通solidity，熟悉 etherjs react 及工具库的开发。

**Comment by Bot80926** [comment]
任务2 看了一下只剩 25讲 没人认领， 可以单独认领 这一讲吗？

**Reply by AmazingAng** [reply]
25讲中使用的工具还没兼容v6，所以不用审核。

**Reply by Bot80926** [reply]
了解

**Comment by chaoticpunk** [comment]


**Reply by AmazingAng** [reply]
填这个报名表，没找到你的贡献者信息: https://docs.google.com/forms/d/e/1FAIpQLSe4KGT8Sh6sJ7hedQRuIYirOoZK_85miz3dw7vA1-YjodgJ-A/viewform?usp=sf_link

**Reply by AmazingAng** [reply]
已发放

**Comment by jsnbyxyjsj** [comment]
报名黑客松。

了解 solidity，熟悉 etherjs react 及工具库的开发。
